packages/web-forms/src/components/common/map/useMapInteractions.ts

```diff
@@ -0,0 +1,205 @@
+		if ('IntersectionObserver' in window) {
+			currentLocationObserver.value = new IntersectionObserver(
+				([entry]) => {
+					if (!entry.isIntersecting) {
+						onMapNotVisible();
+					}
+				},
+				{ root: null, threshold: 0 }
+			);
+			currentLocationObserver.value.observe(mapContainer);
+		}
+	};
+
+	const removeMapInteractions = () => {
+		toggleSelectEvent(false);
+		removeLongPressPoint();
+		removeFeatureDrag();
+	};
+
+	const setCursorPointerForSelect = (event: MapBrowserEvent) => {
+		if (event.dragging || !mapInstance) {
+			return;
+		}
+
+		const hit = mapInstance.hasFeatureAtPixel(event.pixel, {
+			layerFilter: (layer) => layer instanceof WebGLVectorLayer,
+		});
+
+		mapInstance.getTargetElement().style.cursor = hit ? 'pointer' : '';
+	};
+
+	const onSelectInMap = (
+		event: MapBrowserEvent,
+		onSelect?: (feature: Feature | undefined) => void
+	): void => {
+		const hitFeatures = mapInstance.getFeaturesAtPixel(event.pixel, {
+			layerFilter: (layer) => layer instanceof WebGLVectorLayer,
+		});
+
+		const featureToSelect = hitFeatures?.length ? (hitFeatures[0] as Feature) : undefined;
+		if (onSelect) {
+			onSelect(featureToSelect);
+		}
+	};
+
+	const toggleSelectEvent = (
+		bindClick: boolean,
+		onSelect?: (feature: Feature | undefined) => void
+	) => {
+		const onClick = (event: MapBrowserEvent) => onSelectInMap(event, onSelect);
+		mapInstance.un('click', onClick);
+		mapInstance.un('pointermove', setCursorPointerForSelect);
+
+		if (bindClick) {
+			mapInstance.on('click', onClick);
+			mapInstance.on('pointermove', setCursorPointerForSelect);
+		}
+	};
+
+	const setupLongPressPoint = (
+		featuresSource: VectorSource,
+		onLongPress: (feature: Feature) => void
+	) => {
+		if (pointerInteraction.value) {
+			return;
+		}
+
+		let timer: TimerID | null = null;
+		let startPixel: Pixel | null = null;
+		const pixelTolerance = 5;
+
+		pointerInteraction.value = new PointerInteraction({
+			handleDownEvent: (event) => {
+				startPixel = event.pixel;
+				mapInstance.getTargetElement().style.cursor = 'pointer';
+				if (timer) {
+					clearTimeout(timer);
+				}
+
+				timer = setTimeout(() => {
+					if (!startPixel || !timer) {
+						return false;
+					}
+
+					if (!featuresSource.isEmpty()) {
+						featuresSource.clear(true);
+					}
+
+					const feature = new Feature({ geometry: new Point(event.coordinate) });
+					featuresSource.addFeature(feature);
+					onLongPress(feature);
+				}, LONG_PRESS_TIME);
+				return false;
+			},
+			handleMoveEvent: (event) => {
+				if (!startPixel || !timer) {
+					return;
+				}
+
+				const distanceX = Math.abs(event.pixel[0] - startPixel[0]);
+				const distanceY = Math.abs(event.pixel[1] - startPixel[1]);
+				if (distanceX > pixelTolerance || distanceY > pixelTolerance) {
+					clearTimeout(timer);
+					timer = null;
+					startPixel = null;
+					mapInstance.getTargetElement().style.cursor = '';
```

Comment:
Extract a function for this and reuse in the handleUpEvent which (I think) should be identical.